tasmota/berry/drivers/Core200S.be

```diff
@@ -0,0 +1,712 @@
+import string
+import webserver
+
+class StateEnum
+    var value_map
+    var value
+
+    def init(value_map, value)
+        self.value_map = value_map
+        if type(value) == 'int'
+            for k : self.value_map.keys()
+                if self.value_map[k] == value
+                    self.value = value
+                    break
+                end
+            end
+        elif type(value) == 'string'
+            if self.value_map.contains(value)
```

Comment:
you can simply do:
`self.value = value_map.find(value)`

If the key does not exist, it returns `nil`